### ISSUE TITLE
Operator IN doesn't work in MATCH where clause after traverse

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#6830]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2 SNAPSHOT[ ] - .8[x] .rc1[ ] .beta2[ ] .beta1[ ]

## Operating System
- [x] Linux
- [ ] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

## Expected behavior and actual behavior

Expected: for an EmbeddedSet `names`, the `IN` operator checks if an element is contained regardless of the location of the WHERE clause.

Actual: for MATCH statements, the `IN` operator doesn't work and returns no results if used in a WHERE clause after a traversal step. See below for repro -- all MATCH queries were expected to return `#13:0`, but the one with `IN` in the traversal step's WHERE clause returned no results.

## Steps to reproduce the problem
```
CREATE CLASS V1 EXTENDS V
CREATE PROPERTY V1.names EmbeddedSet String
CREATE VERTEX V1 SET names=['foo', 'bar']    // returns #12:0

CREATE CLASS V2 EXTENDS V
CREATE PROPERTY V2.names EmbeddedSet String
CREATE VERTEX V2 SET names=['baz', 'quux']   // returns #13:0

CREATE EDGE E FROM #12:0 TO #13:0

# Returns one result, v2 = #13:0
MATCH {
  class: V1
}.out('E') {
  as: v2
} RETURN $matches

# Returns one result, v2 = #13:0
MATCH {
  class: V2,
  where: ('baz' IN names),
  as: v2
} RETURN $matches

# Returns no results!
MATCH {
  class: V1
}.out('E') {
  where: ('baz' IN names),
  as: v2
} RETURN $matches

# Returns one result, v2 = #13:0
MATCH {
  class: V1
}.out('E') {
  where: (names CONTAINS 'baz'),
  as: v2
} RETURN $matches
```

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

I just fixed it in 2.2.x, the fix will be released in 2.2.13

Thanks

Luigi

**Comment by obi1kenobi** [comment]
Great, thank you!
